tooling/main.go

```diff
@@ -0,0 +1,124 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"fmt"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strings"
+
+	log "github.com/sirupsen/logrus"
+)
+
+
+const (
+	// defaultKubebuilderPath is the fixed value defined in the controller-runtime to looking for the
+	// binaries.
+	defaultKubebuilderPath = "/usr/local/kubebuilder/bin"
+
+	// kubernetesVersion defines the version used to download the binaries
+	kubernetesVersion = "1.16.4"
+)
+
+func main() {
+	log.Infof("Checking EnvTest binaries")
+
+	if hasBinaries(defaultKubebuilderPath) {
+		log.Infof("EnvTest binaries found in : ", defaultKubebuilderPath)
+		os.Exit(0)
+	}
+
+	currentPath, err := os.Getwd()
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	binPath := filepath.Join(currentPath, "bin")
```

Comment:
have a `bin path` option to configure this.

also, we might (not quite sure, what do other folks think?) want to consider installing into $GOBIN with a version suffix, then symlinking to bin, or something, like

```
$GOBIN/kubectl-1.15.6
$PROJECT/.tooling/bin/kubectl -> $GOBIN/kubectl-1.15.6
```

That way, if you're working on a lot of different kb projects (which some folks do), you don't end up with 15 copies of kube-apiserver floating around.

Reply:
IMO the default assets path in controller-runtime ( https://github.com/kubernetes-sigs/controller-runtime/blob/master/pkg/envtest/server.go#L57 ) should be `$GOBIN/` as you suggested instead of `"/usr/local/kubebuilder/bin"` which requires sudo permissions. And then, the project tooling set up in the project when it is not set up with the default path. In this way, users are also able to have diff versions per project as they wish.

Reply:
Raised the issue https://github.com/kubernetes-sigs/controller-runtime/issues/1234 in the controller-runtime